### PR TITLE
[move] Deprecate `spec`s. Remove Move model's dependencies on expansion AST

### PR DESCRIPTION
**Pull request by tnowacki**
## Description 

- Removed the move-model's dependency on the expansion AST, which requires deprecating specs
- Removed spec usage from all files
- Specs will still parse, but be marked as deprecated 

## Test Plan 

- Ran tests

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [X] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### Release notes

Move has deprecated `spec`s as the prover is not in active development

### CODE REVIEW

File: external-crates/move/crates/move-compiler/tests/move_2024/parser/path_usage.move
Comment (cgswords) [inline external-crates/move/crates/move-compiler/tests/move_2024/parser/path_usage.move:1]: Was this brought in from a different diff?

File: external-crates/move/crates/move-compiler/tests/move_2024/expansion/meaningless_copy_move.move
Comment (cgswords) [inline external-crates/move/crates/move-compiler/tests/move_2024/expansion/meaningless_copy_move.move:1]: Was this brought in from a different diff?

File: external-crates/move/crates/move-compiler/tests/move_2024/parser/path_usage.move
Comment (tnowacki) [reply external-crates/move/crates/move-compiler/tests/move_2024/parser/path_usage.move:1]: Yeah its stacked on #15267, which has been stuck by a borked test

File: external-crates/move/crates/bytecode-interpreter-testsuite/Cargo.toml [hunk 1]
```diff
@@ -1,26 +0,0 @@
-[package]
-name = "bytecode-interpreter-testsuite"
```
Comment (tzakian) [inline external-crates/move/crates/bytecode-interpreter-testsuite/Cargo.toml:2]: Does this testsuite need to be deleted?

**Review by tzakian — APPROVED** [review]
LGTM! Just the one question about the interpreter tests ✂️

File: external-crates/move/crates/bytecode-interpreter-testsuite/Cargo.toml (hunk 1 above)
Comment (tnowacki) [reply external-crates/move/crates/bytecode-interpreter-testsuite/Cargo.toml]: I was able to bring it back (except for certain tests). I got a bit excited :)

**Pull request merged**